### ISSUE TITLE
CTRL: A Conditional Transformer Language Model for Controllable Generation

### ISSUE THREAD

**Issue by reo11** [reo11/papers100knock#12]
## 概要
- 条件をprefixとして入力、文章を生成する条件付き言語モデルCTRL
- 文の続き、Wikipediaの記事、批評文、本、Redditの投稿、QA、翻訳など生成可能
- trandformersやsalesforceのgithubから利用可能

## 投稿年/会議
2019年/ArXiv
- https://arxiv.org/abs/1909.05858

## 著者/所属機関
Nitish Shirish Keskar, Bryan McCann, Lav R. Varshney, Caiming Xiong, Richard Socher
- Salesforce Research

## 参考リンク
- https://huggingface.co/transformers/model_doc/ctrl.html
- https://github.com/salesforce/ctrl
- https://twitter.com/kyoun/status/1171894533795147776

## その他メモ
- パラメータ数1.63B
- pretraining data 140GB
  - Wikipedia, OpenWebText, news data, Amazon Reviews, WMT, QA pairs, ...
- 訓練時からcontrol codeを指定し、学習を行なう
- アーキテクチャはtransformer

#### 生成について
- Transformerの最終層が以下のような形式になっており、最終的な出力はVocabのサイズ次元のベクトルである
![image](https://user-images.githubusercontent.com/38539915/80778913-f33d4e00-8ba4-11ea-90c8-d13fcbf7bb7f.png)

- 訓練時はCross Entropy Loss、生成時はSoftmaxした出力を得る
- [ソース](https://github.com/salesforce/ctrl/blob/06e03a633aabe661eddf3bacbceb4a33d6627dbd/pytorch_generation.py#L185)を参照すると、生成の時は生成したトークンを後ろにつけながら指定の長さになるまで生成し続ける模様
